pebbles/models.py

```diff
@@ -515,23 +516,72 @@ def __init__(self, lock_id):
 
 class NamespacedKeyValue(db.Model):
     """ Stores key/value pair data, separated by namespaces
+        This model should be initialized by providing namespace and key as mandatory arguments.
+        It is highly recommended to have a schema for the JSON value field,
+        and provide it during model initialization.
     """
     __tablename__ = 'namespaced_keyvalues'
 
     namespace = db.Column(db.String(32), primary_key=True)
     key = db.Column(db.String(128), primary_key=True)
     _value = db.Column(db.Text)
+    _schema = db.Column(db.Text)
     created_ts = db.Column(db.Float)
     updated_ts = db.Column(db.Float)
 
-    def __init__(self, namespace, key):
+    def __init__(self, namespace, key, schema=None):
         self.namespace = namespace
         self.key = key
+        self.schema = schema
+
+    @classmethod
+    def str_to_bool(cls, val):
+        """ Convert the string into boolean.
+            Useful when value comes from UI and becomes True even if False
+            By default, this function shall return False
+        """
+        if val:
+            val = val.lower()
+            if val in ('true', u'true', '1'):
+                return True
+        return False
+
+    @hybrid_property
+    def schema(self):
+        return load_column(self._schema)
+
+    @schema.setter
+    def schema(self, schema):
+        self._schema = json.dumps(schema)
 
     @hybrid_property
     def value(self):
-        return load_column(self._value)  # Return the json string itself, as the UI has a textfield for modifying it
+        return load_column(self._value)
 
     @value.setter
     def value(self, val):
+        if self.schema:
```

Comment:
As decided earlier, we could implement it as soon as we have a complex driver config coming in. The only thing needed is recursion, and then this should work IMO